wled00/FX_fcn.cpp

```diff
@@ -230,16 +231,28 @@ CRGBPalette16 &Segment::loadPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
   switch (pal) {
     case 0: //default palette. Exceptions for specific effects above
       targetPalette = PartyColors_p; break;
-    case 1: //periodically replace palette with a random one. Doesn't work with multiple FastLED segments
-      if (millis() - _lastPaletteChange > 5000 /*+ ((uint32_t)(255-intensity))*100*/) {
+    case 1: {//periodically replace palette with a random one. Transition palette change in 500ms
+      uint32_t timeSinceLastChange = millis() - _lastPaletteChange;
```

Comment:
millis() returns unsigned log.does this fit into int32? Maybe better use unsigned long, to avoid overflows when computing timesincelastchange. Also cross-check if t_lastpalettechange is ulong.

Reply:
Difference should always be less than uint32_t.